### PR TITLE
[Staking][GUI] Unlock wallet for staking dialogue on click

### PR DESCRIPTION
**Pull request by Rock-N-Troll**
Related to pull request: https://github.com/Veil-Project/veil/pull/913 (and mentioned there as well. This is the separated pull request)

image

 https://github.com/Veil-Project/veil/pull/913#issuecomment-814352629

### CODE REVIEW

**Comment by WetOne** [comment]
ack b8294ee2801aa4088c54a995c194be1ef0a8bf48

**Comment by codeofalltrades** [comment]
What is supposed to do? I clicked placed, the wallet seems for function the same.

**Comment by Rock-N-Troll** [comment]
Clicking the toggle bottom left of screen will also open the unlock wallet password dialog as shown in the screenshot where as previously it would only show the notification that you should unlock wallet for staking

**Review by codeofalltrades — APPROVED** [review]
ACK b8294ee2801aa4088c54a995c194be1ef0a8bf48

File: src/qt/veil/veilstatusbar.cpp [hunk 1]
```diff
@@ -119,27 +119,41 @@ void VeilStatusBar::onCheckStakingClicked(bool res) {
...
+		openToastDialog(dialogMsg, mainWindow);
+		fBlockNextStakeCheckSignal = true;
+		ui->checkStaking->setChecked(false);
+		setStakingText();
+
+		mainWindow->showHide(true);
+		UnlockPasswordDialog *dialog = new UnlockPasswordDialog(/*fUnlockForStakingOnly*/true, this->walletModel, mainWindow);
+		if(openDialogWithOpaqueBackground(dialog, mainWindow, 4)){
+			openToastDialog("Wallet unlocked for staking", mainWindow);
+		}else{
```
Comment (CaveSpectre11) [inline src/qt/veil/veilstatusbar.cpp:141]: NIT: Follow consistent spacing to rest of the module (space between code and brackets
```
if () {
} else {
```
etc...

**Pull request merged**